### DISCUSSION TITLE
How to use alternative S3 providers?

### DISCUSSION THREAD

**Discussion by sotomski** [serverpod/serverpod#1989] [Q&A]
Hey,

I wanted to use an S3 compatible storage hosted somewhere-else-than-aws™️. `S3CloudStorage` looked somewhat promissing and it almost works if not for the fact that it uses: 
- `AwsS3Uploader` which has a hardcoded endpoint `https://$bucket.s3-$region.amazonaws.com`
- `AwsS3Client` which does have a `host` ctor param but it's not used and defaults to `s3.$region.amazonaws.com`

Is there a way to use `S3CloudStorage` with something other than AWS?
Other than copy-pasting the classes above and slightly modifying the AWS-specific bits?

**Comment by SandPod** [comment]
Hello!

Serverpod offers two integrations with cloud storage, these are unfortunately specialized for their respective platform (AWS and GCP).

However, the interface these integrations use is built with the intent to be replaceable. So it is possible to provide your own cloud provider integration and utilize it.

So the current recommended way to communicate with a different provider is to build your own integration.
